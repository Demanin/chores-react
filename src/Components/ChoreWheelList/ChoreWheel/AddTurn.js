import React, { Component } from 'react';
import { Button, FormGroup, FormControl } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class AddTurn extends Component
{
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);

    this.state = {
      value: ''
    };
  }

  handleChange(event) {
    this.props.onSaveClick(event.target.value)
  }

  render() {
    if (!this.props.choreWheel.allowAddTurn) {
      return (
        <Button
          variant="success"
          onClick={this.props.onEditClick}
        >
         <FontAwesomeIcon icon="plus" /> Add New Turn
        </Button>
      );
    }

    return (
      <div>
        <FormGroup
          controlId={"turn-adder-" + this.props.id}
          size="sm"
        >
          <FormControl
            as="select"
            onChange={this.handleChange}
          >
            <option key="-1" value="-1">Select a user</option>
            {this.props.userList.map((user, index) =>
              <option key={index} value={user.id}>{user.name}</option>
            )}
          </FormControl>
        </FormGroup>
      </div>
    );
  }
}

export default AddTurn;
