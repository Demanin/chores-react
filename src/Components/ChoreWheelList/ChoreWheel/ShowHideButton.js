import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';

class ShowHideButton extends Component
{
  render() {
    if (this.props.choreWheel.isVisible) {
      return (
        <Button
          variant="link"
          onClick={() => this.props.onHideClick(this.props.choreWheel.id)}
        >
          <FontAwesomeIcon icon="eye" />
        </Button>
      );
    }

    return (
      <Button
        variant="link"
        onClick={() => this.props.onShowClick(this.props.choreWheel.id)}
      >
        <FontAwesomeIcon icon="eye-slash" />
      </Button>
    );
  }
}

ShowHideButton.propTypes = {
  choreWheel: PropTypes.shape({
    isVisible: PropTypes.bool,
    id: PropTypes.number,
  }),
  onHideClick: PropTypes.func,
  onShowClick: PropTypes.func,
};

export default ShowHideButton;
