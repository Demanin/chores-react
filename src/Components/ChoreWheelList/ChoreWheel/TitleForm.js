import React, { Component } from 'react';
import { FormGroup, FormControl } from 'react-bootstrap';

class TitleForm extends Component
{
  constructor(props) {
    super(props);

    this.handleChange  = this.handleChange.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);

    this.state = {
      value: props.title
    };
  }

  handleChange(event) {
    this.setState({ value: event.target.value });
  }

  handleKeyDown(event) {
    if (event.keyCode === 13) {
      this.props.onSave(this.state.value);
    }
  }

  render() {
    return (
      <FormGroup
        controlId={"title-editor-" + this.props.id}
      >
        <FormControl
          autoFocus
          value={this.state.value}
          onChange={this.handleChange}
          onBlur={() => this.props.onSave(this.state.value)}
          onKeyDown={this.handleKeyDown}
        />
      </FormGroup>
    );
  }
}

export default TitleForm;
