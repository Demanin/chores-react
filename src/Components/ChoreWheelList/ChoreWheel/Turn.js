import React, { Component } from 'react';
import { Card, Row, Col, Button, ButtonGroup } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

class Turn extends Component
{
  render() {
    if (!this.props.turn) {
      return null;
    }

    const hasNextTurn = (!!this.props.nextTurn);
    const hasThirdTurn = (!!this.props.thirdTurn);

    return (
      <div>
        <Card body size="sm" style={hasNextTurn ? {marginBottom: '0px'} : {}}>
          <Row>
            <Col xs={6} md={9}>
              <h4 style={{marginTop: '0px', marginBottom: '0px'}}>
                {this.props.turn}
              </h4>
            </Col>
            <Col xs={6} md={3} className="text-right">
              <ButtonGroup>
                <Button
                  variant="success"
                  size="xs"
                  onClick={this.props.onCompleteClick}
                >
                  <FontAwesomeIcon icon="check" />
                </Button>
                <Button
                  variant="warning"
                  size="xs"
                  onClick={this.props.onSkipClick}
                >
                  <FontAwesomeIcon icon="forward" />
                </Button>
                <Button
                  variant="danger"
                  size="xs"
                  onClick={this.props.onRemoveClick}
                >
                  <FontAwesomeIcon icon="trash" />
                </Button>
              </ButtonGroup>
            </Col>
          </Row>
        </Card>
        {hasNextTurn &&
          <div className="text-info well-sm">
            {this.props.nextTurn}
          </div>
        }
        {hasThirdTurn &&
          <div className="well-sm" style={{color: '#aaa'}}>
            {this.props.thirdTurn}
          </div>
        }
      </div>
    );
  }
}

export default Turn;
