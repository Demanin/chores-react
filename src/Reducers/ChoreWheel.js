
const ChoreWheel = (state = {}, action) => {
  switch (action.type) {
    case 'ALLOW_ADD_TURN':
      return {
        ...state,
        allowAddTurn: true,
      };
    case 'EDIT_WHEEL_TITLE':
      return {
        ...state,
        title: {
          ...state.title,
          isEditable: true,
        },
      };
    case 'SAVE_WHEEL_TITLE':
      return {
        ...state,
        title: {
          text: action.title,
          isEditable: false,
        },
      };
    default:
      return state;
  }
};

export default ChoreWheel;
