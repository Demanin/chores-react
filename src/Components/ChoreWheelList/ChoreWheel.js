import { Button, Card, Col, Row } from 'react-bootstrap';
import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import ShowHideButton from 'Components/ChoreWheelList/ChoreWheel/ShowHideButton';
import VisibleTurnList from 'Containers/VisibleTurnList';
import WheelHeader from 'Containers/WheelHeader';
import WheelTurnAdder from 'Containers/WheelTurnAdder';

class ChoreWheel extends Component
{
  render() {
    return (
      <Card className="mt-4">
        <Card.Body>
          <Row>
            <Col xs={9} md={10} style={{minHeight: '40px'}}>
              <WheelHeader
                choreWheel={this.props.choreWheel}
              />
            </Col>
            <Col xs={3} md={2} className="text-right">
              <Button
                variant="danger"
                size="xs"
                onClick={() => this.props.onRemoveWheelClick(this.props.choreWheel.id)}
              >
                <FontAwesomeIcon icon="trash" />
              </Button>
            </Col>
          </Row>

          <VisibleTurnList
            choreWheel={this.props.choreWheel}
          />
        </Card.Body>
        <Card.Footer>
          <Row>
            <Col xs={9} md={10}>
              <WheelTurnAdder
                choreWheel={this.props.choreWheel}
              />
            </Col>
            <Col xs={3} md={2} className="text-right">
              <ShowHideButton
                onShowClick={this.props.onShowClick}
                onHideClick={this.props.onHideClick}
                choreWheel={this.props.choreWheel} />
            </Col>
          </Row>
        </Card.Footer>
      </Card>
    );
  }
}

ChoreWheel.propTypes = {
  choreWheel: PropTypes.shape({
    id: PropTypes.number,
    turnList: PropTypes.arrayOf(
      PropTypes.shape({ userId: PropTypes.number }),
    ),
    title: PropTypes.shape({
      text: PropTypes.string,
      isEditable: PropTypes.bool,
    }),
    isVisible: PropTypes.bool,
    priority: PropTypes.number,
  }),
  onShowClick: PropTypes.func,
  onHideClick: PropTypes.func,
  onRemoveWheelClick: PropTypes.func,
};

export default ChoreWheel;
