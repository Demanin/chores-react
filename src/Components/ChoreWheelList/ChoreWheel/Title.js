import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TitleForm from 'Components/ChoreWheelList/ChoreWheel/TitleForm';

class Title extends Component
{
  render() {
    if (!this.props.choreWheel.title.isEditable) {
      return (
        <h3
          style={{marginTop: '0px'}}
          onClick={this.props.onEditClick}
        >
          {this.props.choreWheel.title.text}
        </h3>
      );
    }

    return (
      <TitleForm
        choreWheel={this.props.choreWheel}
        title={this.props.choreWheel.title.text}
        onSave={this.props.onSaveClick}
      />
    );
  }
}

Title.propTypes = {
  choreWheel: PropTypes.shape({
    id: PropTypes.number,
    turnList: PropTypes.arrayOf(
      PropTypes.shape({ userId: PropTypes.number }),
    ),
    title: PropTypes.shape({
      text: PropTypes.string,
      isEditable: PropTypes.bool,
    }),
    isVisible: PropTypes.bool,
    priority: PropTypes.number,
  }),
  onEditClick: PropTypes.func,
  onSaveClick: PropTypes.func,
};

export default Title;
