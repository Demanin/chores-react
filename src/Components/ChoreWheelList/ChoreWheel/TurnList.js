import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Turn from './Turn';

class TurnList extends Component
{
  render() {
    if (this.props.turnList.length === 0) {
      return null;
    }

    return (
      <div>
        {this.props.turnList.map((turn, index) =>
          <Turn
            key={index}
            turn={turn}
            onClick={() => this.props.onClick(this.props.choreWheelId, index)}
          />
        )}
      </div>
    );
  }
}

TurnList.propTypes = {
  turnList: PropTypes.arrayOf(PropTypes.string),
  onClick: PropTypes.func,
  choreWheelId: PropTypes.number,
};

export default TurnList;
