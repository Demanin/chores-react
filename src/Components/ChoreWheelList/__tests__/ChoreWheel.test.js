import ChoreWheel from 'src/Components/ChoreWheelList/ChoreWheel';
import React from 'react';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('ChoreWheel', () => {
  let onShowClick;
  let onHideClick;
  let onRemoveWheelClick;

  beforeEach(() => {
    onShowClick = jest.fn();
    onHideClick = jest.fn();
    onRemoveWheelClick = jest.fn();
  });

  test('empty wheel snapshot', () => {
    const fixture = shallow(<ChoreWheel
      choreWheel={{}}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('WheelHeader has choreWheel as prop', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ChoreWheel
      choreWheel={{ id }}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    expect(fixture.find('Connect(Title)').prop('choreWheel')).toEqual({ id });
  });

  test('remove wheel click', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ChoreWheel
      choreWheel={{ id }}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    fixture.find('Button').simulate('click');

    expect(onRemoveWheelClick).toHaveBeenCalledWith(id);
  });

  test('VisibleTurnList has choreWheel as prop', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ChoreWheel
      choreWheel={{ id }}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    expect(fixture.find('Connect(TurnList)').prop('choreWheel')).toEqual({ id });
  });

  test('WheelTurnAdder has choreWheel as prop', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ChoreWheel
      choreWheel={{ id }}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    expect(fixture.find('Connect(AddTurn)').prop('choreWheel')).toEqual({ id });
  });

  test('ShowHideButton props', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ChoreWheel
      choreWheel={{ id }}
      {...{ onShowClick, onHideClick, onRemoveWheelClick }}
    />);

    const expected = {
      choreWheel: { id },
      onShowClick,
      onHideClick,
    };
    expect(fixture.find('ShowHideButton').props()).toEqual(expected);
  });
});
