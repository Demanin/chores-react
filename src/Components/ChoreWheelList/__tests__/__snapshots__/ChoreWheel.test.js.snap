// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoreWheel empty wheel snapshot 1`] = `ShallowWrapper {}`;
